### ISSUE TITLE
tutorial 0 + tutorial 1 password and jwt issue

### ISSUE THREAD

**Issue by vks2** [PostgREST/postgrest#1687]
Am I right with this: if i implement your example in tutorial 0 and 1 I would have an REST API service open for anyone, because of 
create role web_anon **nologin**;
If I proceed with Tutorial 1 and do it step-by-step I still do not require any password when call it from localhost:3000 
could you please share all tutorial.conf used in Tutorial 1, not only this: jwt-secret = "<the password you made>"
because it's an REST api open for anyone if you have your IPv and port 3000 open to public.
Am I wrong with smth?
Thanks much.

**Comment by wolfgangwalther** [comment]
> because it's an REST api open for anyone if you have your IPv and port 3000 open to public.

Yes, if you open this port to the internet, everyone can access it. This is **not** a problem per-se. Anonymous requests are possible and that's ok.

If you really want to block all anonymous requests before they even reach PostgREST, you could add a reverse proxy in front of it.



> Am I right with this: if i implement your example in tutorial 0 and 1 I would have an REST API service open for anyone, because of
> create role web_anon **nologin**;

I am not sure why you highlighted `nologin` here, the `nologin` part is really not important at all. In this context it does not mean anything. `nologin` prevents a direct login on the database with this user - but I am sure you will not expose your database port to the public. `nologin` does still allow PostgREST to **switch to that user** with `SET ROLE`. This is what PostgREST does for anonymous requests.

Now everything is about which kind of privileges you give to the `web_anon` user. You can just remove all postgres privileges (no schema access, no table access, no whatever) from this user - then the anonymous user can not do anything with your API.



> If I proceed with Tutorial 1 and do it step-by-step I still do not require any password when call it from localhost:3000

You don't need any authentication for the anonymous user. But you **need** authentication for trusted users. In tutorial 1, this is the `todo_user`. To be able to access the api with this user, you need to provide a JWT. The JWT is your "password". The JWT is signed with the `jwt-secret` that only the server knows and not the public.

You might want to read up more in the "Authentication" section of the docs: https://postgrest.org/en/latest/auth.html

In short: It all depends on which privileges you give to your database users.

**Comment by steve-chavez** [comment]
Transferring this one to docs :arrow_right: :page_facing_up:
